snuba/utils/streams/abstract.py

```diff
@@ -138,6 +138,24 @@ def poll(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def tell(self) -> Mapping[TStream, TOffset]:
+        """
+        Return the read offsets for all assigned streams.
```

Comment:
Do you plan to specify the behavior of the api at this level for consumer groups with no committed offsets available on the broker?

Reply:
Yes, every partition will have an absolute read offset that is exposed via this API, even if they do not have any committed offset (or have even been consumed from yet.) The how and/or why might be a little more apparent after reading in #528.

(You may have already picked up that I am going to need to rebase this change on top of that one.)